core/src/View/Kokkos_BasicView.hpp

```diff
@@ -348,15 +354,24 @@ class BasicView {
           "Constructing View and initializing data with uninitialized "
           "execution space");
     }
-    return data_handle_type(Impl::make_shared_allocation_record<ElementType>(
-        arg_mapping.required_span_size(),
-        Impl::get_property<Impl::LabelTag>(prop_copy),
-        Impl::get_property<Impl::MemorySpaceTag>(prop_copy),
-        has_exec ? std::optional<execution_space>{Impl::get_property<
-                       Impl::ExecutionSpaceTag>(prop_copy)}
-                 : std::optional<execution_space>{std::nullopt},
-        std::integral_constant<bool, alloc_prop::initialize>(),
-        std::integral_constant<bool, alloc_prop::sequential_host_init>()));
+    if constexpr (has_exec) {
+      return data_handle_type(Impl::make_shared_allocation_record<ElementType>(
+          arg_mapping.required_span_size(),
+          Impl::get_property<Impl::LabelTag>(prop_copy),
+          Impl::get_property<Impl::MemorySpaceTag>(prop_copy),
+          std::make_optional(
+              Impl::get_property<Impl::ExecutionSpaceTag>(prop_copy)),
+          std::bool_constant<alloc_prop::initialize>(),
+          std::bool_constant<alloc_prop::sequential_host_init>()));
+    } else {
+      return data_handle_type(Impl::make_shared_allocation_record<ElementType>(
+          arg_mapping.required_span_size(),
+          Impl::get_property<Impl::LabelTag>(prop_copy),
+          Impl::get_property<Impl::MemorySpaceTag>(prop_copy),
+          std::optional<execution_space>{},
+          std::bool_constant<alloc_prop::initialize>(),
+          std::bool_constant<alloc_prop::sequential_host_init>()));
+    }
```

Comment:
This also seems unrelated. Can you spell out why it is included here?

Reply:
Because we need it for the View implementation, and we want to make the PR introducing View smaller.